docs/getting_started/typescript/hello_world_in_typescript/index.md

```diff
@@ -88,142 +158,155 @@ Create the `package.json` file in the root of your project and add the following
 
-Your Activity Definition can accept input parameters. Review the [Activity parameters](https://docs.temporal.io/dev-guide/typescript/foundations/#activity-parameters) section of the Temporal documentation for more details, as there are some limitations you'll want to be aware of when running more complex applications.
+The `proxyActivities` method is also where you set options for how Temporal works with Activities. In this example, you have specified that the Start-to-Close Timeout for your Activity will be one minute, meaning that your Activity has one minute to begin before it times out. Of all the Temporal timeout options, `startToCloseTimeOut` is the one you should always set.
 
-Create the file `activities.ts` in the `src` directory:
+You also set the Retry Policy for Activities this way. In this example, you're using the default Retry Policy values, but by leaving it in you have a more clear picture of what happens. Note that the `maximumAttempts` is commented out, which means there's no limit to the number of times Temporal will retry your Activities if they fail.
 
-Add the following code to define a `greet` function:
+With the imports and options in place, you can define the Workflow itself. In the TypeScript SDK, you implement a Workflow the same way you define an Activity; using an exportable async TypeScript function. Add the following code to call both Activities, using the value of the first as the input to the second:
 
-<!--SNIPSTART typescript-hello-activity-->
-[hello-world/src/activities.ts](https://github.com/temporalio/samples-typescript/blob/main/hello-world/src/activities.ts)
 ```ts
-export async function greet(name: string): Promise<string> {
-  return `Hello, ${name}!`;
+// The Temporal Workflow.
+export async function getAddressFromIP(name: string): Promise<string> {
+
+  try {
+    const ip = await getIP();
+    try {
+      const location = await getLocationInfo(ip);
+      return `Hello, ${name}. Your IP is ${ip} and your location is ${location}`;
+    } catch (e) {
+      throw new workflow.ApplicationFailure("Failed to get location");
+    }
+  } catch (e) {
+    throw new workflow.ApplicationFailure("Failed to get IP");
+  }
+
 }
 ```
-<!--SNIPEND-->
 
-You've completed the logic for the application; you have a Workflow and an Activity defined. Next, you'll write code to configure and launch a Worker.
+This code uses a `try/catch` block, but because you've configured unlimited retries, there won't be any exceptions caught. However, if you change the Retry Policy's maximum retries, or you specify specific non-retryable exceptions, this code will be in place to handle those errors.
```

Comment:
```suggestion
This code uses a `try/catch` block, but because you've configured unlimited retries, there won't be any exceptions caught. However, if you change the Retry Policy's maximum retries, or you specify non-retryable exceptions, this code will be in place to handle those errors.
```